uploadNeuroDB/NeuroDB/File.pm

```diff
@@ -355,20 +386,47 @@ sub setParameter {
     $this->{'parameters'}->{$paramName} = $value;
 
     if($this->getFileDatum('FileID')) {
-	my $fileID = $this->getFileDatum('FileID');
-	my $paramID = $this->getParameterTypeID($paramName);
-	my $query = "SELECT count(*) AS counter FROM parameter_file WHERE FileID=$fileID AND ParameterTypeID=$paramID";
-	my $sth = ${$this->{'dbhr'}}->prepare($query);
-	$sth->execute();
-	my $row = $sth->fetchrow_hashref();
-
-	$value = ${$this->{'dbhr'}}->quote($value);
-	if($row->{'counter'} > 0) {
-	    $query = "UPDATE parameter_file SET Value=$value, InsertTime=UNIX_TIMESTAMP() WHERE FileID=$fileID AND ParameterTypeID=$paramID";
-	} else {
-	    $query = "INSERT INTO parameter_file SET Value=$value, FileID=$fileID, ParameterTypeID=$paramID, InsertTime=UNIX_TIMESTAMP()";
-	}
-	${$this->{'dbhr'}}->do($query);
+	    my $fileID = $this->getFileDatum('FileID');
+	    my $paramID = $this->getParameterTypeID($paramName);
+	    (my $query = <<QUERY) =~ s/\n/ /gm; 
+    SELECT 
+        count(*) AS counter 
+    FROM 
+        parameter_file 
+    WHERE 
+        FileID=? 
+        AND ParameterTypeID=?
+QUERY
+
+	    my $sth = ${$this->{'dbhr'}}->prepare($query);
+	    $sth->execute($fileID, $paramID);
+	    my $row = $sth->fetchrow_hashref();
+
+	    $value = ${$this->{'dbhr'}}->quote($value);
+	    if($row->{'counter'} > 0) {
+            ($query = <<QUERY) =~ s/\n/ /gm;
+    UPDATE
+        parameter_file
+    SET 
+        Value=?,
+        InsertTime=UNIX_TIMESTAMP()
+    WHERE
+        FileID=?
+        AND ParameterTypeID=?
+QUERY
+	    } else {
+            ($query = <<QUERY) =~ s/\n/ /gm;
+    INSERT INTO
+        parameter_file 
+    SET
+        Value=?, 
+        FileID=?, 
+        ParameterTypeID=?, 
+        InsertTime=UNIX_TIMESTAMP()
+QUERY
+	    }
+        $sth = ${$this->{'dbhr'}}->prepare($query);
+        $sth->execute($value, $fileID, $paramID);
```

Comment:
@cmadjar,

(this execute failed error appears on line 977 in MRI.pm, it seems to be the case when you replaced do($query)???)
DBD::mysql::st execute failed: Cannot add or update a child row: a foreign key constraint fails (`mouna_db`.`parameter_file`, CONSTRAINT `FK_parameter_file_1` FOREIGN KEY (`FileID`) REFERENCES `files` (`FileID`)) at /data/loris-MRI/bin/mri/uploadNeuroDB/NeuroDB/File.pm line 429.